clippy_lints/src/operators/numeric_arithmetic.rs

```diff
@@ -15,7 +15,10 @@ pub struct Context {
     const_span: Option<Span>,
 }
 impl Context {
-    fn skip_expr(&mut self, e: &hir::Expr<'_>) -> bool {
+    fn skip_expr<'tcx>(&mut self, cx: &LateContext<'tcx>, e: &'tcx hir::Expr<'_>) -> bool {
+        if is_from_proc_macro(cx, e) {
```

Comment:
This should be the last check here. i.e. `self.expr_id.is_some() || self.const_span.map_or(false, |span| span.contains(e.span)) || is_from_proc_macro(cx, e)`

Reply:
Done